app/screens/route-list/route-list-screen.tsx

```diff
@@ -40,7 +41,11 @@ export const RouteListScreen = observer(function RouteListScreen({ navigation, r
 
   useEffect(() => {
     if (trains.isSuccess) {
-      setRouteData((prevData) => [...prevData, routePlan.date.toDateString(), ...trains.data])
+      setRouteData((prevData) =>
+        uniqBy([...prevData, routePlan.date.toDateString(), ...trains.data], (item) => {
+          return typeof item === "string" ? item : item.trains.map((train) => train.departureTimeString).join()
```

Comment:
Unique by train numbers, not departure times as they can be the same

`item.trains.map((train) => train.trainNumber).join()`